code/modules/hydroponics/hydroponics.dm

```diff
@@ -75,6 +75,7 @@
 	plant_hud_set_water()
 
 /obj/machinery/hydroponics/Destroy()
+	remove_from_all_data_huds()
```

Comment:
I thought I answered you on this; you properly used procs; there were a few instances of using procs you did miss, but its effects were negligible, at best, no even noticeable by players, at worse.

I think this is the only actual noteworthy mistake, which was preventing hydro trays from garbage collecting.